### ISSUE TITLE
investigate mismatch of input flux with galsim galaxy image flux

### ISSUE THREAD

**Issue by ismael-mendoza** [prob-ml/bliss#511]
Is it because of the long tails of the DeVaucauleur component? should we renormalize the image ?

**Comment by dereklhansen** [comment]
Is this just because the galaxy size is finite? In other words, if we make the galaxy size arbitrary large, will the flux converge?

If this is the case, summing up the 53x53 pixels to calculate flux will always lead to an underestimate. It might be useful to look into how coadd estimates fluxes. They may do some kind of upward adjustment to account for the finite number of pixels.

**Comment by dereklhansen** [comment]
To test this hypothesis, we can generate a synthetic catalog with Galsim galaxies and see how much we underestimate flux.

**Comment by ismael-mendoza** [comment]
Here is what I found from digging into the `galsim` side of things (the `galsim.GSObject.drawImage` function is thankfully well documented). 

Galsim actually mentions that the flux might not be captured fully when drawing a finite image and it actually suggests a size for the image based on this and a threshold. See excerpt of the documentation below

```
If drawImage() will be creating the image from scratch for you, then there are several ways
to control the size of the new image.  If the ``nx`` and ``ny`` keywords are present, then
an image with these numbers of pixels on a side will be created.  Similarly, if the ``bounds``
keyword is present, then an image with the specified bounds will be created.  Note that it
is an error to provide an existing `Image` when also specifying ``nx``, ``ny``, or
``bounds``.  In the absence of ``nx``, ``ny``, and ``bounds``, drawImage will decide a good
size to use based on the size of the object being drawn.  Basically, it will try to use an
area large enough to include at least 99.5% of the flux.

NOTE: This value 0.995 is really ``1 - folding_threshold``.  You can change the value of
``folding_threshold`` for any object via `GSParams`.

(...)

On return, the image will have an attribute ``added_flux``, which will be set to the total
flux added to the image.  This may be useful as a sanity check that you have provided a
large enough image to catch most of the flux.  For example::

    >>> obj.drawImage(image)
    >>> assert image.added_flux > 0.99 * obj.flux
```

I've tested this in the code using some typical values we might expect of HLR and Flux in the images we are simulating. It does seem to be only a problem with the DeVaucaulers profile in our case (due to the long tails). In the code below you can see in the first example galsim suggested to produce an image of size 178x178 (vs what we use in the code 53x53). The problem will become worse the higher the flux of the galaxy, as expected.

![Screen Shot 2022-04-01 at 1 31 21 PM](https://user-images.githubusercontent.com/11745764/161313326-34181316-ecfd-474c-9a4f-ee2fd674fab1.png)

**Comment by ismael-mendoza** [comment]
I'm not sure what part (if any) of the training pipeline would experience problems due to this, since BLISS will just be trained to reconstruct whatever pixels for a given galaxy are in an image. 

I suppose that if our padded tiles do not encompass enough of a real SDSS galaxy with a DeVacauleur component we might return an underestimate of the flux of that galaxy which makes sense.  I would say that making the padded tiles ~170x170 sounds like overkill. 

> If this is the case, summing up the 53x53 pixels to calculate flux will always lead to an underestimate. It might be useful to look into how coadd estimates fluxes. They may do some kind of upward adjustment to account for the finite number of pixels.

This sounds like a good idea.

**Comment by ismael-mendoza** [comment]
After some searches in the SDSS website and the PHOTO [paper](https://www.astro.princeton.edu/~rhl/photomisc/deblender.pdf) I could not find information about this. Do you know @jeff-regier ? or where to look? We could also ask Robert since he would probably know. 

Perhaps if BLISS targets the true flux directly (like with stars) it could learn to do this adjustment if it turns out to be necessary.

**Comment by dereklhansen** [comment]
What would it look like if you recreated the figure from Wednesday, but comparing the magnitude of PHOTO galaxies vs COADD galaxies?

**Comment by dereklhansen** [comment]
My main concern wasn't about Galsim specifically, but rather that our method for estimating fluxes has a negative bias because we sum over a finite range of pixels that don't cover everything.

> Perhaps if BLISS targets the true flux directly (like with stars) it could learn to do this adjustment if it turns out to be necessary.

I think this makes sense, and it more closely matches how other catalogs calculate flux. This is an except from https://arxiv.org/pdf/1504.00452.pdf describing how they estimate galaxy fluxes and shapes in SExtractor. They estimate flux by backing it out from the best-fit bulge-disk model.
![image](https://user-images.githubusercontent.com/12701492/161321006-e72cfbc5-f9f7-4e97-a70b-1f79a99452f3.png)

**Comment by jeff-regier** [comment]
> To test this hypothesis, we can generate a synthetic catalog with Galsim galaxies and see how much we underestimate flux.

Testing the hypothesis sounds like a good. My hunch is that our underestimates of galaxy flux are too large to be explained by the fraction of the galaxy flux that falls outside of a 53x53 subimage.

**Comment by ismael-mendoza** [comment]
The mismatch with coadd gets resolved in #579, this [slide](https://docs.google.com/presentation/d/1nVg7OHu2E9gCO_uxaqLkid4QdICkyHDagprcHtUJ_0s/edit#slide=id.gf3ed96089f_0_10) from our meeting checks that we don't incur in a significant bias ignoring the flux outside outside the postage stamp of our individual galaixes. So indeed this issue is closed :)
